A/AWS_MQTT/build_tarballs.jl

```diff
@@ -0,0 +1,139 @@
+    # https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-gettickcount64
+    export CXXFLAGS=-D_WIN32_WINNT=0x0600
+fi
+
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_STATIC_LIBS=ON \
+    -DBUILD_SHARED_LIBS=OFF \
+    -GNinja \
+    ..
+ninja -j${nproc}
+ninja install
+
+cd $WORKSPACE/srcdir/s2n-tls
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc}
+cmake --install .
+
+cd $WORKSPACE/srcdir/aws-c-common
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-cal
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-io
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-compression
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-http
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-mqtt
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    FileProduct(["lib64/libaws-c-mqtt.a"], :libawsmqtt),
```

Comment:
If at the end you get only a static library, how are you going to use it from Julia?  If you're planning to use it from Julia.

Reply:
You are right, I will have to make a shared library at the end of this. I haven't figured out how to configure CMake to do that, yet. Simply configuring the aws-c-mqtt build to output a shared library didn't work.

Reply:
If I build all of these as shared libraries, that will give me the same trouble as in #4727, right? So my goal is to build the dependencies of aws-c-mqtt as static libraries, and build aws-c-mqtt as a shared library.

Reply:
Yeah, I guess you want a shared library only for AWS MQTT